flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/DataMapManager.java

```diff
@@ -1066,18 +1079,39 @@ public Map<String, String> getVisualizableProperties(SemanticModel sm, JsonEleme
             if (symbol.getName().isEmpty() || !symbol.getName().get().equals(type)) {
                 continue;
             }
-            TypeDefinitionSymbol typeDefSymbol = (TypeDefinitionSymbol) symbol;
-            TypeSymbol typeSymbol = typeDefSymbol.typeDescriptor();
-            TypeSymbol rawTypeSymbol = CommonUtils.getRawType(typeSymbol);
-            TypeDescKind kind = rawTypeSymbol.typeKind();
-            if (isEffectiveRecordType(kind, rawTypeSymbol)) {
-                if (kind == TypeDescKind.ARRAY || isArray) {
-                    return Map.of("expression", "[]");
-                } else if (kind == TypeDescKind.RECORD) {
-                    return Map.of("expression", "{}");
-                }
+            Map<String, String> expressionMap = getExpressionMapForType(symbol, isArray);
+            if (!expressionMap.isEmpty()) {
+                return expressionMap;
+            }
+            throw new IllegalStateException("Unsupported type for visualizable properties: " +
+                    CommonUtils.getRawType(((TypeDefinitionSymbol) symbol).typeDescriptor()).typeKind());
+        }
+
+        String[] typeSegments = type.split(":");
+        String prefix = typeSegments.length > 1 ? typeSegments[0] : "";
+        String typeName = typeSegments.length > 1 ? typeSegments[1] : typeSegments[0];
+        Symbol matchedSymbol = getMatchedSymbol(prefix, typeName, semanticModel);
```

Comment:
Here, `getMatchedSymbol` has little bit heavy loops and this function runs for all the types. Shall we improve this implementation to run only for type-references. I would check whether a particular type is type-ref and has module-prefix before this heavy loop.